### PR TITLE
Add build from source to setup.py

### PR DESCRIPTION
**Pull request by kdheepak**

### CODE REVIEW

File: helics_apps-pip/setup.py [hunk 1]
```diff
@@ -1,40 +1,131 @@
...
 
 # Get the directory setup.py is located in
 setup_py_dir = os.path.abspath(os.path.dirname(__file__))
 
 # Get the README.md contents
-README_contents = io.open(os.path.join(setup_py_dir, 'README.md'), encoding="utf-8").read()
+README_contents = io.open(os.path.join(setup_py_dir, "README.md"), encoding="utf-8").read()
+
+HELICS_SOURCE = os.path.join(setup_py_dir, "./_source")
+HELICS_VERSION = versioneer.get_version()
```
Comment (nightlark) [inline helics_apps-pip/setup.py:28]: The versioneer version is the Python package version with post/pre tags, and devN for commits made after a tagged/pypi release (e.g. `v2.7.0.post.dev1`). This won't work with the additional Python-related tags for downloading a HELICS source tarball.

Comment (kdheepak) [reply helics_apps-pip/setup.py:28]: Fixed in https://github.com/GMLC-TDC/helics-packaging/pull/11/commits/e1b79f809026a7ae54534c92f49890f3917e341b.

**Review by nightlark — APPROVED** [review]

**Comment by nightlark** [comment]
Wheels uploaded to test.pypi look good

**Comment by nightlark** [comment]
Ah though I guess we'll need to make it upload an sdist now too.

**Comment by kdheepak** [comment]
I think this will make no difference to the wheels the way it is set up now, right? It using prebuilt binaries for the wheels based on the previous setup. But if you try to install using `--no-binary` it’ll build from source.

**Comment by kdheepak** [comment]
Oh, true. We’ll need the sdist now.

**Pull request merged**